packages/documentation/plugins/remark/gatsby-remark-mdx-mermaid/remark-mermaid/src/utils.js

```diff
@@ -0,0 +1,92 @@
+const crypto = require('crypto');
+const fs = require('fs-extra');
+const path = require('path');
+const which = require('npm-which')(process.cwd());
+const execSync = require('child_process').execSync;
+
+const PLUGIN_NAME = 'remark-mermaid';
+
+/**
+ * Accepts the `source` of the graph as a string, and render an SVG using
+ * mermaid.cli. Returns the path to the rendered SVG.
+ *
+ * @param  {string} source
+ * @param  {string} destination
+ * @return {string}
+ */
+function render(source, destination) {
+  const staticFilePath = `${process.cwd()}/static/mermaid-svgs`;
+  const unique = crypto.createHmac('sha1', PLUGIN_NAME).update(source).digest('hex');
+  const mmdcExecutable = which.sync('mmdc');
+  const mmdPath = path.join(staticFilePath, `${unique}.mmd`);
+  const svgFilename = `${unique}.svg`;
+  const svgPath = path.join(staticFilePath, svgFilename);
+  // Write temporary file
+  fs.outputFileSync(mmdPath, source);
+
+  // Invoke mermaid.cli
+  execSync(`${mmdcExecutable} -i ${mmdPath} -o ${svgPath} -b transparent`);
+
+  // Clean up temporary file
+  fs.removeSync(mmdPath);
+
+  return { source, imgSrc: `/mermaid-svgs/${svgFilename}`};
```

Comment:
this returns the path to the image we will use on our page. Source gives us back an object regarding what mermaid has changed.